build.yaml

```diff
@@ -169,6 +162,105 @@ common_cli:
       fi
     }
 
+    # Retries a command if it fails, using the exponential backoff algorithm.
+    # Usage:
+    # retry_n_times_with_timeout <max retries> <timeout in seconds> command
+    #
+    function retry_n_times_with_timeout {
+      set +e
+      max_tries=\${1}
+      seconds=\${2}
+      shift 2
+
+      let delay=15
+      let tries=0
+      while [ \${tries} -lt \${max_tries} ]; do
+        echo "executing...  \${*}"
+        if (timeout --foreground --signal=SIGKILL \${seconds} bash -c "\${*}"); then
+            break
```

Comment:
What error message will we see if this timeout gets exceeded? Is it something that will help someone who isn't familiar with this code to figure out what's going on? 

If not, please add an error message.